src/rgw/rgw_op.h

```diff
@@ -545,6 +546,10 @@ class RGWCreateBucket : public RGWOp {
 public:
   RGWCreateBucket() : has_cors(false) {}
 
+  void emplace_attr(std::string&& key, buffer::list&& bl) {
+    attrs.emplace(key, bl); /* key and bl are r-value refs */
```

Comment:
confusing, but `key` and `bl` are lvalues here.. so you need to re-`move()` them:
`attrs.emplace(std::move(key), std::move(bl));`

i found this to be a decent explanation as to why: [Is an Rvalue Reference an Rvalue?](http://thbecker.net/articles/rvalue_references/section_05.html)